src/System.Memory/tests/ParsersAndFormatters/Parser/TestData.Parser.Date.cs

```diff
@@ -12,7 +13,32 @@ public static IEnumerable<ParserTestData<DateTime>> DateTimeParserTestData
         {
             get
             {
-                foreach (ParserTestData<DateTime> testData in DateTimeFormatterTestData.ToParserTheoryDataCollection())
+				// Format J tests for DateTime parsing.
+				yield return new ParserTestData<DateTime>("0997-07-16", DateTime.Parse("0997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16", DateTime.Parse("1997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20", DateTime.Parse("1997-07-16T19:20"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30", DateTime.Parse("1997-07-16T19:20:30"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45", DateTime.Parse("1997-07-16T19:20:30.45"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555", DateTime.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+                
+                // Test fraction rounding.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45555554", DateTime.Parse("1997-07-16T19:20:30.45555554"), 'J', expectedSuccess: true);
+                // DateTime.Parse will round up to 7dp in this case, but we expect parser to truncate.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.45555555", DateTime.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+
+				// Test with timezone designator.
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555Z", DateTime.ParseExact("1997-07-16T19:20:30.4555555Z", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555+01:00", DateTime.ParseExact("1997-07-16T19:20:30.4555555+01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTime>("1997-07-16T19:20:30.4555555-01:00", DateTime.ParseExact("1997-07-16T19:20:30.4555555-01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+
+				// Invalid strings.
+				yield return new ParserTestData<DateTime>("997-07-16", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTime>("1997-07", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTime>("1997-07-6", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTime>("1997-07-16Z", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTime>("1997-07-16+01:00", default, 'J', expectedSuccess: false);
```

Comment:
We should add other invalid strings. For example:
`1997-7-06`
`1997-07-16T`
`1997-07-6T01`